docs/root/configuration/advanced/well_known_filter_state.rst

```diff
@@ -13,9 +13,16 @@ The following list of filter state objects are consumed by Envoy extensions:
    * - **Filter state key**
      - **Purpose**
    * - ``envoy.tcp_proxy.cluster``
-     - :ref:`TCP proxy <config_network_filters_tcp_proxy>` dynamic cluster selection on a per-connection basis.
+     - | :ref:`TCP proxy <config_network_filters_tcp_proxy>` dynamic cluster name selection
+       | on a per-connection basis.
    * - ``envoy.network.transport_socket.original_dst_address``
      - | :ref:`Original destination cluster <arch_overview_load_balancing_types_original_destination>` dynamic address selection.
        | Fields:
        | - *ip*: IP address value;
        | - *port*: port value.
+   * - ``envoy.upstream.dynamic_host``
```

Comment:
if we're adding docs should we add release notes as well?

Reply:
It's not a new feature, it's mostly about documenting and exposing existing features better. I can add a note if you feel strongly about it.